discovery-infra/tests/test_auth.py

```diff
@@ -0,0 +1,282 @@
+        
+        return assisted_service_api.create_client(url, offline_token, **kwargs)
+
+
+    yield get_api_client
+
+@pytest.mark.skipif(not env_variables['offline_token'], reason="not cloud env")
+class TestAuth(BaseTest):
+    @pytest.fixture()
+    def cluster(self):
+        clusters = []
+        api_clients = []
+        def get_cluster_func(api_client, cluster_name: Optional[str] = None):
+            api_clients.append(api_client)
+            if not cluster_name:
+                cluster_name = random_name()
+
+            res = api_client.create_cluster(cluster_name,
+                                            ssh_public_key=env_variables['ssh_public_key'],
+                                            openshift_version=env_variables['openshift_version'],
+                                            pull_secret=env_variables['pull_secret'],
+                                            base_dns_domain=env_variables['base_domain'],
+                                            vip_dhcp_allocation=env_variables['vip_dhcp_allocation'])
+            clusters.append(res)
+            return res
+
+        yield get_cluster_func
+        for client in api_clients:
+            for cluster in clusters:
+                with suppress(ApiException):
+                    client.delete_cluster(cluster.id)
+
+    def _send_dummy_step_result(self, api_client, cluster_id, host_id):
+        api_client.host_post_step_result(
+            cluster_id, 
+            host_id,
+            step_type="inventory",
+            step_id="inventory-e048e0db",
+            exit_code=0,
+            output="null"
+        )
+
+    @pytest.mark.regression
+    def test_user_authorization_negative(self, api_client, node_controller, cluster):
+        client_user1 = api_client()
+        client_user2 = api_client(offline_token=env_variables['second_offline_token'])
+
+        user1_cluster_id = cluster(client_user1, env_variables['cluster_name']).id
+
+        #user2 cannot get user1's cluster 
+        self.assert_http_error_code(
+            api_client=client_user2,
+            api_call=self.get_cluster,
+            status=404,
+            reason="Not Found",
+            cluster_id=user1_cluster_id
+        )
+
+        #user2 cannot delete user1's cluster
+        self.assert_http_error_code(
+            api_client=client_user2,
+            api_call=self.delete_cluster,
+            status=404,
+            reason="Not Found",
+            cluster_id=user1_cluster_id
+        )
+
+        #user2 cannot generate ISO user1's cluster
+        self.assert_http_error_code(
+            api_client=client_user2,
+            api_call=self.generate_and_download_image,
+            status=404,
+            reason="Not Found",
+            cluster_id=user1_cluster_id
+        )
+
+        self.generate_and_download_image(cluster_id=user1_cluster_id, api_client=client_user1)
```

Comment:
i think we have this part in every test we add. please lets move all those thing to cluster object and reuse